### PR TITLE
Deployment Scripts

### PR DESCRIPTION
**Pull request by sirrus233**
Adds some lovely helper scripts, now that we're ON THE INTERNET.

A deployment script for frontend and backend respectively, and a provisioning script to do all the first-time setup required if we ever swap out the EC2 instance.

There are some implicit requirements to run these:

`provision.sh`:
 * Need credentials to ssh to the host as the `ec2-user`

`deploy.sh` (Backend flavor)
* Requires cabal
* Same ssh permissions that are required to provision

`deploy.sh` (Frontend flavor)
* Requires npm
* Requires aws-cli
* Requires access to the relevant AWS resources (s3+cloudfront), stored under a "wotrcommunity" profile

### CODE REVIEW

File: infrastructure/lib/infrastructure-stack.ts [hunk 1]
```diff
@@ -11,8 +11,10 @@ export class InfrastructureStack extends cdk.Stack {
     constructor(scope: Construct, id: string, props?: cdk.StackProps) {
         super(scope, id, props);
 
+        const elasticIp = new ec2.CfnEIP(this, "ElasticIP");
```
Comment (sirrus233) [inline infrastructure/lib/infrastructure-stack.ts:14]: Re-org'd this so we can swap out / kill the instance but keep the same IP.

File: backend/bin/provision.sh [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+
+BIN_DIR=$(dirname "$0")
+source "${BIN_DIR}/config.sh"
+  
+ssh "$SERVER_USER@$SERVER_HOST" <<EOF
+  set -e
+  
+  mkdir -p wotr-server
+  
+  echo "Acquiring SSL certificate..."
```
Comment (chelseaSchmidt) [inline backend/bin/provision.sh:13]: I tried to figure out a script for automating a letsencrypt SSL cert renewal at one point in my life and remember just giving up in despair lol, this script is sweet and I will probably reference it in the future

File: frontend/bin/deploy.sh [hunk 3]
```diff
@@ -0,0 +1,22 @@
...
+
+echo "Building the frontend..."
+npm install
+npm run build
+
+echo "Uploading files to S3..."
+aws --profile wotrcommunity s3 sync $BUILD_DIR s3://$S3_BUCKET --delete
+
+echo "Invalidating CloudFront cache..."
+aws --profile wotrcommunity cloudfront create-invalidation --distribution-id $CLOUDFRONT_DISTRIBUTION_ID --paths "/*"
```
Comment (chelseaSchmidt) [inline frontend/bin/deploy.sh:20]: So convenient ✨

**Review by chelseaSchmidt — APPROVED** [review]
👏

File: backend/bin/provision.sh (hunk 2 above)
Comment (sirrus233) [reply backend/bin/provision.sh:13]: Lemme know if you wanna chat about any of the UNIX-y stuff!

Comment (chelseaSchmidt) [reply backend/bin/provision.sh:13]: 🙏

**Pull request merged**